File: content/LLD/Reference-Code/2. SOLID Principles/5. Dependency Inversion Principle (DIP)/1. Bad Example/notification_service.py
from email_service import EmailService
from sms_service import SMSService


class NotificationService:
    def __init__(self):
        self.email_service = EmailService()
        self.sms_service = SMSService()

    def notifyByEmail(self, message):
        self.email_service.send_email(message)

    def notifyBySMS(self, message):
        self.sms_service.send_sms(message)


ns = NotificationService()
ns.notifyByEmail("Good morning")
ns.notifyBySMS("Hey")

File: content/LLD/Reference-Code/2. SOLID Principles/5. Dependency Inversion Principle (DIP)/1. Bad Example/NotificationService.java
public class NotificationService {
    private EmailService emailService;
    private SMSService smsService;

    public NotificationService() {
        this.emailService = new EmailService();
        this.smsService = new SMSService();
    }

    public void notifyByEmail(String message) {
        this.emailService.sendEmail(message);
    }

    public void notifyBySMS(String message) {
        this.smsService.sendSMS(message);
    }

    public static void main(String[] args) {
        NotificationService ns = new NotificationService();
        ns.notifyByEmail("Good morning");
        ns.notifyBySMS("Hey");
    }
}
